mailpoet/assets/js/src/segments/dynamic/templates/index.tsx

```diff
@@ -64,72 +55,63 @@ export function SegmentTemplates(): JSX.Element {
     <div className="mailpoet-main-container">
       <HideScreenOptions />
       <TopBarWithBeamer />
-      <Flex
-        className="mailpoet-heading"
-        direction={['column', 'row'] as any} // eslint-disable-line @typescript-eslint/no-explicit-any -- typed as string but supports string[] and this is needed to make the component responsive
-        gap="16px"
+      <PageHeader
+        className="mailpoet-templates-header"
+        heading={__('Start with a pre-built segment', 'mailpoet')}
+        headingPrefix={
+          <BackButton
+            href="#/"
+            label={__('Segments list', 'mailpoet')}
+            aria-label={__('Navigate to the segments list page', 'mailpoet')}
+          />
+        }
       >
-        <FlexBlock>
-          <h1 className="wp-heading-inline">
-            <Button
-              icon={chevronLeft}
-              aria-label={__('Navigate to the segments list page', 'mailpoet')}
-              href="#/"
-              label={__('Segments list', 'mailpoet')}
-            />
-            {__('Start with a pre-built segment', 'mailpoet')}
-          </h1>
-        </FlexBlock>
-
-        <FlexItem>
-          <SearchControl label={__('Search segment templates', 'mailpoet')} />
-        </FlexItem>
-
-        <FlexItem>
-          <Button
-            variant="secondary"
-            href={`#${ROUTES.NEW_DYNAMIC_SEGMENT}`}
-            data-automation-id="new-custom-segment"
-            onClick={(): void => {
-              trackNewCustomSegment();
-            }}
-          >
-            {__('Create custom segment', 'mailpoet')}
-          </Button>
-        </FlexItem>
-      </Flex>
+        <SearchControl label={__('Search segment templates', 'mailpoet')} />
+        <Button
+          variant="secondary"
+          href={`#${ROUTES.NEW_DYNAMIC_SEGMENT}`}
+          data-automation-id="new-custom-segment"
+          onClick={() => void trackNewCustomSegment()}
+        >
+          {__('Create custom segment', 'mailpoet')}
+        </Button>
+      </PageHeader>
 
       {errors.length > 0 && (
         <APIErrorsNotice errors={errors.map((error) => ({ message: error }))} />
       )}
 
       <TabPanel tabs={tabs}>
         {(tab) => (
-          <div className="mailpoet-templates-card-grid">
+          <Grid>
             {templates
               .filter(
                 (template) =>
                   tab.name === 'all' || template.category === tab.name,
               )
               .map((template) => (
-                <TemplateListItem key={template.name} template={template} />
+                <Item
+                  name={template.name}
+                  description={template.description}
+                  category={getCategoryNameBySlug(template.category)}
+                  badge={template.isEssential ? 'essential' : undefined}
+                  onClick={() => void createFromTemplate(template)}
+                />
```

Comment:
On the segments template page I'm getting

```
On segments page:
Warning: Each child in a list should have a unique "key" prop.
Check the render method of `TabPanel`. See https://reactjs.org/link/warning-keys for more information.
...
```

I think these `Item`s need a `key` to prevent it.

Reply:
Thanks, fixed in https://github.com/mailpoet/mailpoet/pull/5275/commits/84bc0e543c22cfae40433ef3188e37688a226bf8.